.github/workflows/blocks-e2e.yml

```diff
@@ -0,0 +1,179 @@
+
+            - name: Upload artifacts on failure
+              if: ${{ failure() }}
+              uses: actions/upload-artifact@v3.1.2
+              with:
+                  name: e2e-with-gutenberg-test-report-${{matrix.part}}
+                  path: reports/e2e
+
+            - name: Archive flaky tests report
+              uses: actions/upload-artifact@v3.1.2
+              if: always()
+              with:
+                  name: flaky-tests-report-${{ matrix.part }}
+                  path: flaky-tests
+                  if-no-files-found: ignore
+
+    JSE2ETests:
+        name: JavaScript E2E Tests (latest)
+        strategy:
+            fail-fast: false
+            matrix:
+                part: [1, 2, 3, 4, 5]
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Cache node_modules
+              id: cache-node-modules
+              uses: actions/cache@v3
+              env:
+                  cache-name: cache-node-modules
+              with:
+                  path: node_modules
+                  key: ${{ runner.os }}-modified-build-${{ hashFiles('**/package-lock.json') }}
+                  restore-keys: |
+                      ${{ runner.os }}-modified-build-${{ env.cache-name }}-
+                      ${{ runner.os }}-modified-build-
+                      ${{ runner.os }}-modified-
+
+            - name: Setup node version and npm cache
+              uses: actions/setup-node@v4
+              with:
+                  node-version-file: '.nvmrc'
+                  cache: 'npm'
+
+            - name: Install Node dependencies
+              if: steps.cache-node-modules.outputs.cache-hit != 'true'
+              run: npm install --no-optional --no-audit
+
+            - name: Build Assets
+              run: FORCE_REDUCED_MOTION=true npm run build
+
+            - name: blocks.ini setup
+              run: |
+                  echo -e 'woocommerce_blocks_phase = 3\nwoocommerce_blocks_env = tests' > blocks.ini
+            - name: Get Composer Cache Directory
+              id: composer-cache
+              run: |
+                  echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+            - uses: actions/cache@v3
+              with:
+                  path: ${{ steps.composer-cache.outputs.dir }}
+                  key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+                  restore-keys: |
+                      ${{ runner.os }}-composer-
+
+            - name: Set up PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: '8.0'
+                  coverage: none
+                  tools: composer
+
+            - name: Composer install
+              run: |
+                  composer install
```

Comment:
I'm 99% sure that virtually _all_ of this can be replaced by using `setup-woocommerce-monorepo` 😄